module/zfs/arc.c

```diff
@@ -2305,7 +2315,11 @@ add_reference(arc_buf_hdr_t *hdr, void *tag)
 			arc_evictable_space_decrement(hdr, state);
 		}
 		/* remove the prefetch flag if we get a reference */
+		if (HDR_HAS_L2HDR(hdr))
+			l2arc_hdr_arcstats_update(hdr, B_FALSE, B_TRUE);
```

Comment:
The boolean flags make these calls a bit difficult to decipher. Maybe some defines for the four flavors?
```C
#define l2arc_hdr_arcstats_increment(hdr) l2arc_hdr_arcstats_update((hdr), B_TRUE, B_FALSE)
#define l2arc_hdr_arcstats_decrement(hdr) l2arc_hdr_arcstats_update((hdr), B_FALSE, B_FALSE)
#define l2arc_hdr_arcstats_increment_state(hdr) l2arc_hdr_arcstats_update((hdr), B_TRUE, B_TRUE)
#define l2arc_hdr_arcstats_decrement_state(hdr) l2arc_hdr_arcstats_update((hdr), B_FALSE, B_TRUE)
```